bundle/edu.gemini.wdba.xmlrpc.server/src/main/java/edu/gemini/wdba/tcc/ObservationEnvironment.java

```diff
@@ -157,15 +152,16 @@ public TargetEnvironment getTargetEnvironment() {
         return _targetEnv;
     }
 
+    public SortedSet<GuideProbe> usedGuiders() {
+        return _targetEnv.getGuideEnvironment().getPrimaryReferencedGuiders();
+    }
+
     public boolean containsTargets(GuideProbe probe) {
-        final Option<GuideProbeTargets> gtOpt = _targetEnv.getPrimaryGuideProbeTargets(probe);
-        return gtOpt.exists(GuideProbeTargets::containsTargets);
```

Comment:
This is a key difference from the previous version.  Instead of just checking whether there are targets associated with a guide probe, now require the guide probe to be associated with targets that are configured to be actually _used_.  This is what fixes REL-2789.